File: redome-courier/src/main/java/br/org/cancer/redome/courier/model/Transportadora.java
package br.org.cancer.redome.courier.model;

import java.io.Serializable;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.validation.constraints.NotNull;

import org.hibernate.annotations.Where;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty.Access;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Builder.Default;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Entidade que representa a transportadora que irá realizar o transporte do
 * material coletado (empresa terceirizada de courier).
 * 
 * @author Pizão.
 */
@Entity
@SequenceGenerator(name = "SQ_TRAN_ID", sequenceName = "SQ_TRAN_ID", allocationSize = 1)
@Table(name = "TRANSPORTADORA")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class Transportadora implements Serializable {
	private static final long serialVersionUID = -56406142749782205L;

	@Id
	@Column(name = "TRAN_ID")
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_TRAN_ID")
	private Long id;

	@Column(name = "TRAN_NOME")
	@NotNull
	private String nome;

	@Column(name = "TRAN_ATIVO")
	@Default
	private Boolean ativo = true;
		
	@OneToOne(mappedBy = "transportadora", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
	@Where(clause="ENCO_IN_EXCLUIDO = 0")
	@NotNull
	private EnderecoContatoTransportadora endereco;

	
	@OneToMany(mappedBy = "transportadora", cascade = CascadeType.ALL)
	@Where(clause="EMCO_IN_EXCLUIDO = 0")
	@NotNull
	private List<EmailContatoTransportadora> emails;
	
	@OneToMany(mappedBy = "transportadora", cascade = CascadeType.ALL)
	@Where(clause="COTE_IN_EXCLUIDO = 0")
	@NotNull
	private List<ContatoTelefonicoTransportadora> telefones;
	
	@Transient
	@JsonProperty(access = Access.WRITE_ONLY)
	private List<Long> usuarios;
	
	
}

File: redome-front/src/app/shared/model/transportadora.ts
import { BaseEntidade } from '../base.entidade';
import { Usuario } from '../dominio/usuario';
import { ContatoTelefonicoTransportadora } from './contato.telefonico.transportadora';
import { EmailContatoTransportadora } from './email.contato.transportadora';
import { EnderecoContatoTransportadora } from './endereco.contato.transportadora';

/**
 * @author Pizão
 *
 * Entidade que representa as transportadores parceiras responsáveis
 * por levar o material coletado até o destino (CT).
 */
export class Transportadora extends BaseEntidade {
	private _id: number;
	private _nome: string;
	private _usuarios: Usuario[] = [];
	private _endereco:EnderecoContatoTransportadora;
	private _emails:EmailContatoTransportadora[] = [];
	private _telefones:ContatoTelefonicoTransportadora[] = [];


	public get id(): number {
		return this._id;
	}

	public set id(value: number) {
		this._id = value;
	}

	public get nome(): string {
		return this._nome;
	}

	public set nome(value: string) {
		this._nome = value;
	}

	/**
	 * Getter usuarios
	 * @return {Usuario[]}
	 */
	public get usuarios(): Usuario[] {
		return this._usuarios;
	}

	/**
	 * Setter usuarios
	 * @param {Usuario[]} value
	 */
	public set usuarios(value: Usuario[]) {
		this._usuarios = value;
	}

    /**
     * Getter endereco
     * @return {EnderecoContatoTransportadora}
     */
	public get endereco(): EnderecoContatoTransportadora {
		return this._endereco;
	}

    /**
     * Setter endereco
     * @param {EnderecoContatoTransportadora} value
     */
	public set endereco(value: EnderecoContatoTransportadora) {
		this._endereco = value;
	}


    /**
     * Getter emails
     * @return {EmailContatoTransportadora[] }
     */
	public get emails(): EmailContatoTransportadora[]  {
		return this._emails;
	}

    /**
     * Setter emails
     * @param {EmailContatoTransportadora[] } value
     */
	public set emails(value: EmailContatoTransportadora[] ) {
		this._emails = value;
	}

    /**
     * Getter telefones
     * @return {ContatoTelefonicoTransportadora[] }
     */
	public get telefones(): ContatoTelefonicoTransportadora[]  {
		return this._telefones;
	}

    /**
     * Setter telefones
     * @param {ContatoTelefonicoTransportadora[] } value
     */
	public set telefones(value: ContatoTelefonicoTransportadora[] ) {
		this._telefones = value;
	}


	public jsonToEntity(res: any): Transportadora {
		let transportadora: Transportadora = new Transportadora();
		transportadora.id = res.id;
		transportadora.nome = res.nome;
		transportadora.usuarios = [];
		if(res.usuarios){
			res.usuarios.forEach(u=> {
				transportadora.usuarios.push(new Usuario().jsonToEntity(u));
			});
		}
		if(res.endereco){
			transportadora.endereco = new EnderecoContatoTransportadora().jsonToEntity(res.endereco);
		}
		if(res.emails){
			res.emails.forEach(e=> {
				transportadora.emails.push(new EmailContatoTransportadora().jsonToEntity(e));
			});
		}
		if(res.telefones){
			res.telefones.forEach(t=> {
				transportadora.telefones.push(new ContatoTelefonicoTransportadora().jsonToEntity(t));
			});
		}
		return transportadora;
	}
}
